File: backend/foorrng/src/main/java/com/d205/foorrng/user/repository/UserRepository.java
package com.d205.foorrng.user.repository;

import com.d205.foorrng.user.entity.FavoriteFood;
import com.d205.foorrng.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {


    Optional<User> findByUserUid(Long userUid);


    Optional<User> findByEmail(String email);

    Optional<User> findByName(String name);

    Optional<User> findByBusinessNumber(String businessNumber);
}

File: android/foorrng_owner/domain/src/main/java/com/tasteguys/foorrng_owner/domain/repository/UserRepository.kt
package com.tasteguys.foorrng_owner.domain.repository

import com.tasteguys.foorrng_owner.domain.model.user.LoginData

interface UserRepository {
    suspend fun registerOwner(
        userUid: Long,
        name: String,
        email: String
    ): Result<Long>

    suspend fun login(
        userUid: Long,
        name: String,
        email: String
    ): Result<LoginData>

    suspend fun registBusinessNumber(
        businessNumber: String
    ): Result<String>
}